File: event-schemas/data/csharp/Product.cs
namespace entities
{
  using System.Collections.Generic;

  public class Product
  {
    private string productId;
    private string sku;
    private string category;
    private string name;
    private string brand;
    private string variant;
    private double? quantity;
    private double? price;
    private string currency;
    private string imageUri;
    private string uri;
    private Dictionary<string, dynamic> additionalProperties;

    public string ProductId 
    {
      get { return productId; }
      set { productId = value; }
    }

    public string Sku 
    {
      get { return sku; }
      set { sku = value; }
    }

    public string Category 
    {
      get { return category; }
      set { category = value; }
    }

    public string Name 
    {
      get { return name; }
      set { name = value; }
    }

    public string Brand 
    {
      get { return brand; }
      set { brand = value; }
    }

    public string Variant 
    {
      get { return variant; }
      set { variant = value; }
    }

    public double? Quantity 
    {
      get { return quantity; }
      set { quantity = value; }
    }

    public double? Price 
    {
      get { return price; }
      set { price = value; }
    }

    public string Currency 
    {
      get { return currency; }
      set { currency = value; }
    }

    public string ImageUri 
    {
      get { return imageUri; }
      set { imageUri = value; }
    }

    public string Uri 
    {
      get { return uri; }
      set { uri = value; }
    }

    public Dictionary<string, dynamic> AdditionalProperties 
    {
      get { return additionalProperties; }
      set { additionalProperties = value; }
    }
  }
}

File: event-schemas/src/entities/Product.ts


class Product {
  private _productId: string;
  private _sku?: string;
  private _category?: string;
  private _name?: string;
  private _brand?: string;
  private _variant?: string;
  private _quantity?: number;
  private _price?: number;
  private _currency?: string;
  private _imageUri?: string;
  private _uri?: string;
  private _additionalProperties?: Map<String, object | string | number | Array<unknown> | boolean | null>;

  constructor(input: {
    productId: string,
    sku?: string,
    category?: string,
    name?: string,
    brand?: string,
    variant?: string,
    quantity?: number,
    price?: number,
    currency?: string,
    imageUri?: string,
    uri?: string,
  }) {
    this._productId = input.productId;
    this._sku = input.sku;
    this._category = input.category;
    this._name = input.name;
    this._brand = input.brand;
    this._variant = input.variant;
    this._quantity = input.quantity;
    this._price = input.price;
    this._currency = input.currency;
    this._imageUri = input.imageUri;
    this._uri = input.uri;
  }

  get productId(): string { return this._productId; }
  set productId(productId: string) { this._productId = productId; }

  get sku(): string | undefined { return this._sku; }
  set sku(sku: string | undefined) { this._sku = sku; }

  get category(): string | undefined { return this._category; }
  set category(category: string | undefined) { this._category = category; }

  get name(): string | undefined { return this._name; }
  set name(name: string | undefined) { this._name = name; }

  get brand(): string | undefined { return this._brand; }
  set brand(brand: string | undefined) { this._brand = brand; }

  get variant(): string | undefined { return this._variant; }
  set variant(variant: string | undefined) { this._variant = variant; }

  get quantity(): number | undefined { return this._quantity; }
  set quantity(quantity: number | undefined) { this._quantity = quantity; }

  get price(): number | undefined { return this._price; }
  set price(price: number | undefined) { this._price = price; }

  get currency(): string | undefined { return this._currency; }
  set currency(currency: string | undefined) { this._currency = currency; }

  get imageUri(): string | undefined { return this._imageUri; }
  set imageUri(imageUri: string | undefined) { this._imageUri = imageUri; }

  get uri(): string | undefined { return this._uri; }
  set uri(uri: string | undefined) { this._uri = uri; }

  get additionalProperties(): Map<String, object | string | number | Array<unknown> | boolean | null> | undefined { return this._additionalProperties; }
  set additionalProperties(additionalProperties: Map<String, object | string | number | Array<unknown> | boolean | null> | undefined) { this._additionalProperties = additionalProperties; }
}
export default Product;
